MarsRover/RoverClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MarsRover
+{
+    public class RoverClient
+    {
+        private Grid _grid;
+        private IHeading _heading;
+        private List<Obstacle> _obstacles;
+        private Rover _rover;
+        private List<RoverNavigationHistory> _navigationHistory = new List<RoverNavigationHistory>();
+
+
+        public RoverClient()
+        {
+            _grid = new Grid(3, 3, 5, 5);
+            _heading = new NorthHeading();
+            _obstacles = CreateObstacles(3);
+            _rover = new Rover(_grid, _heading, _obstacles);
+            AddToNavigationHistory();
+        }
+
+        
+        public void RoverInstructions()
+        {
+            var commands = string.Empty;
+            DisplayInstructions();
+            
+            while(!commands.Equals("X"))
+            {
+                Console.WriteLine($"Your current position on the surface grid is {_rover.XCoordinate()}, {_rover.YCoodinate()} and the Rover is facing {_rover.Direction()}.\n");
+                Console.WriteLine("Please enter your rover commands: ");
+                commands = Console.ReadLine().ToUpper().Replace(" ", "");
+                _rover = _rover.ExecuteCommands(commands);
```

Comment:
If you get rid of the `return this` and change this line to `_rover.ExecuteCommands(commands);` won't it still work?  The pattern of overriding a variable that you use on the same line stands out to me (and was a source of problems in the Poker kata too if you recall).

Reply:
It sure will.  Not sure why my brain does that at times.  :-)